dynamics_class: "quad_near_hover"

safety_states: 
  - 1
  - 2
  - 4
  - 5
safety_controls:
  - 0
  - 3 


obstacles: 

  obstacle1:
    type: 'Rectangle'
    mode: 'Active'
    indices:
      - 0
      - 1
    minVal:
      - 0.65
      - 0.9
    maxVal:
      - 0.75
      - 1.9
    padding: .2


  obstacle3:
    type: 'Rectangle'
    mode: 'Detection'
    indices:
      - 0
      - 1
    minVal:
      - -2.5
      - 0
    maxVal:
      - -2.0
      - 0.95
    padding: .1
    detectionradius: 1.0
  obstacle4:
    type: 'Circle'
    mode: 'Detection'
    indices:
      - 0
      - 1
    center:
      - -0.5
      - 2.0
    radius: 0.5
    padding: .1
    detectionradius: 0.7
  # obstacle2:
  #   type: 'Rectangle'
  #   mode: 'Service'
  #   indices:
  #     - 0
  #     - 1
  #   minVal:
  #     - -2.0
  #     - 0
  #   maxVal:
  #     - -1.6
  #     - 0.65
  #   padding: .5

  obstacle2:
    type: 'Rectangle'
    mode: 'Update'
    indices:
      - 0
      - 1
    minVal:
      - -2.2
      - 0.0
    maxVal:
      - -1.3
      - 0.65
    padding: .1
    updatetime: 15

boundary:
  padding: 0.1
  indices:
    - 0
    - 1
    - 2
    - 3
  minVal:
    - -2.0
    - -0.1
    - -2.0
    - -5.0
  maxVal:
    - 4.0
    - 3.0
    - 2.0
    - 5.0

obstacle_update_rate_hz: 10

state_domain:
  lo:
    - -2.0
    - -0.1
    - -2.0
    - -5.0
  hi:
    - 4.0
    - 3.0
    - 2.0
    - 5.0
  resolution:
    - 31
    - 31
    - 25
    - 25
  periodic_dims: []
  
control_space:
  n_dims: 2
  lo:
    - -0.1
    - 6.0
  hi: 
    - 0.1
    - 15.0

disturbance_space:
  n_dims: 1
  lo: 
    - 0.0
  hi: 
    - 0.0

actuation_updates: []

disturbance_updates:
  disturbance1:
    n_dims: 1
    lo:
      - 0.0
    hi:
      - 0.0
